client/src/components/admin/AdminRound/AdminRoundContent.jsx

```diff
@@ -76,6 +76,23 @@ function setStoredBatchResults(roundId, results) {
   }
 }
 
+/**
+ * Combine results and batchResults to produce a temporary results-like array
+ * that has all batchResults changes applied.  This allows us to display a
+ * warning if a result is being entered which matches a batched result exactly.
+ */
+function combineResultsAndBatchResults(results, batchResults) {
+  const combinedResults = [...results];
+  batchResults.forEach((batchedResult) => {
+    const i = combinedResults.findIndex((res) => res.id === batchedResult.id);
+    combinedResults[i] = {
+      ...combinedResults[i],
+      attempts: batchedResult.attempts,
+    };
+  });
+  return combinedResults;
```

Comment:
Btw. you can also write it like this:

```suggestion
  return results.map((result) => {
    const batchResult = batchResults.find((batchResult) => batchResult.id === result.id);
    if (batchResult) {
      return { ...result, attempts: batchResult.attempts };
    } else {
      return result;
    }
  );
```

I consider it slightly more direct in terms of readability, though either version is fine :)